### ISSUE TITLE
treat disk_store.realm as cache

### ISSUE THREAD

**Issue by ildar** [vector-im/element-android#5078]
### Your use case

I found that file `disk_store.realm` is 1GB (huge).
The containing data I totally recoverable from server (just tested). Hence it is cache by it's nature.

I'd like this data to be somehow discardable in case of storage shortage.

### Have you considered any alternatives?

manual
> rm Grip:/data/data/im.vector.app/files/*/disk_store.realm

then restart the app.

### Additional context

Storage constrained devices

**Comment by ouchadam** [comment]
I had also noticed this the other day, it seems to be related to the realm compacting which we do on cold app launch, I had a 2gb file compact to 20mb~ after relaunching the app

**Comment by bernankebernard** [comment]
i had this disk_store.realm 6.9gb !!!
on another phone it was up to 4gb

after deleting it has grown up to ±40mb already, and keep growing

**Comment by ouchadam** [comment]
this is a feature of the realm database we're using

> The size of a Realm Database file is always larger than the total size of the objects stored within it. This architecture enables some of realm's great performance, concurrency, and safety benefits.

we automatically compact the database on each cold app launch, 

perhaps we're missing a dedicated storage settings page (similar to signal/whatsapp), with the option to manually compact the database, @daniellekirkwood what do you think?

**Comment by bmarty** [comment]
I agree on the concept to move the session database to the cache for the same reason that you. I already tried to do so.
The main annoying side effect, is that after deleting the cache, a Firebase push will trigger an initial sync and this can take time for big accounts.
We have done some recent improvement to limit the stored history for rooms by deleting the previous chunks in case of gappy sync. It will not guarantee that the database cannot grow up indefinitely but this is better than before.
